service/src/main/java/org/broadinstitute/listener/relay/Utils.java

```diff
@@ -35,6 +46,29 @@ public static boolean isSetCookiePath(URI uri) {
     }
   }
 
+  /**
+   * Whether the request represents user or kernel activity, and should thus trigger a touch of the
+   * runtime to prevent autopause.
+   */
+  public static boolean isKeepAliveRequest(RelayedHttpListenerRequest listenerRequest) {
+    URI uri = listenerRequest.getUri();
+    var splitted = Arrays.asList(uri.getPath().split("/"));
+    boolean isKeepAliveExempt =
+        listenerRequest.getHttpMethod().equals("GET")
+            && KEEP_ALIVE_EXEMPT_PATHS.stream()
+                .anyMatch(
+                    exemptPath -> {
+                      int exemptPathSize = exemptPath.size();
+                      if (splitted.size() != PATH_PREFIX_LENGTH + exemptPathSize) {
```

Comment:
Oh yeah, this does feel a bit brittle indeed, using regex patterns would be best here, if possible i would do it in this PR